src/components/SvgIcon.js

```diff
@@ -75,18 +75,39 @@ export function SvgIcon({ name, className = '', inline = false, title = '' }) {
   );
 }
 
+/**
+ * Register an icon for use with the `SvgIcon` component.
+ *
+ * Returns a symbol that can be passed as the `name` prop to `SvgIcon` in order
+ * to render this icon.
+ *
+ * @param {string|symbol} name - A name for this icon
+ * @param {string} markup - SVG markup for the icon
+ * @return {symbol}
+ */
+export function registerIcon(name, markup) {
+  const key = typeof name === 'string' ? Symbol(name) : name;
+  iconRegistry.set(key, markup);
+  return key;
+}
+
 /**
  * Register icons for use with the `SvgIcon` component.
  *
- * @param {IconMap} icons
- * @param {object} options
+ * @deprecated Prefer the `registerIcon` function instead which will return a
+ * key that does not conflict with existing icons.
```

Comment:
As I mentioned in the PR description, I'm not 100% sold that we do want to deprecate this, but I have an open mind.